deployment/data-streams/changeset/common.go

```diff
@@ -66,3 +68,32 @@ func proposeAllOrNothing(ctx context.Context, oc deployment.OffchainClient, prs
 
 	return proposedJobs, err
 }
+
+// fetchExternalJobID looks for an existing job that matches the given labels and returns its ID.
+// If no job is found, it returns a nil UUID.
+func fetchExternalJobID(e deployment.Environment, nodeIDs []string, selectors []*ptypes.Selector) (externalJobID uuid.UUID, err error) {
+	jobsResp, err := e.Offchain.ListJobs(e.GetContext(), &job.ListJobsRequest{
+		Filter: &job.ListJobsRequest_Filter{
+			NodeIds:   nodeIDs,
+			Selectors: selectors,
+		},
+	})
+	if err != nil {
+		return uuid.Nil, fmt.Errorf("failed to list jobs: %w", err)
+	}
+
+	switch len(jobsResp.Jobs) {
+	case 0:
+		// No job found, return nil UUID
+	case 1:
+		// One job found, return its ID
+		externalJobID, err = uuid.Parse(jobsResp.Jobs[0].Uuid)
+		if err != nil {
+			err = fmt.Errorf("failed to parse external job ID: %w", err)
+		}
+	default:
+		// More than one job found, return error
+		err = fmt.Errorf("multiple jobs found: %d", len(jobsResp.Jobs))
```

Comment:
Currently in the distribute_stream_spec changeset this is non-deterministic: https://github.com/smartcontractkit/chainlink/blob/01f64c979c84e2128b7db1a2809a220d64aad87b/deployment/data-streams/changeset/jd_distribute_llo_jobs.go#L218

Think of this case: 1 node is down when distributing a stream job. That job will be proposed later with a different externalJobID since we'd have called `uuid.New` in a different context. 

We can ask the caller to specify the externalJobId so that we guarantee it's consistent even across re-runs

Reply:
Otherwise we can accept that there will be multiple possible values for externalJobId

Reply:
Hm, that's fair. I will limit `fetchExternalJobID` to one `nodeID`, so the expectation for a unique external job id is a realistic one.